sentry_sdk/scope.py

```diff
@@ -1,69 +1,82 @@
 from .utils import logger
 
 
+def _attr_setter(fn):
+    return property(fset=fn, doc=fn.__doc__)
+
+
 class Scope(object):
     """The scope holds extra information that should be sent with all
     events that belong to it.
     """
 
-    __slots__ = ["_data", "_breadcrumbs", "_event_processors", "_error_processors"]
+    __slots__ = (
+        "_fingerprint",
+        "_transaction",
+        "_user",
+        "_tags",
+        "_contexts",
+        "_extras",
+        "_breadcrumbs",
+        "_event_processors",
+        "_error_processors",
+    )
 
     def __init__(self):
-        self._data = {}
-        self._breadcrumbs = []
-        self._event_processors = []
         self._error_processors = []
```

Comment:
Is it intentional that error processors are not cleared?

Reply:
event and error processors should survive clear